tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +217,224 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
+    createAnchoringManagerAndLazyDocument = ->
+      annotator =
+        publish: sinon.spy()
+
+      am = new Annotator.Plugin.EnhancedAnchoring()
+      am.annotator = annotator
+      am.pluginInit()
+
+      am.documentAccessStrategies.unshift
+        name: "Dummy two-phase"
+        mapper: DummyDocumentAccess
+
+      am.strategies.push
+        name: "dummy anchoring strategy"
+        code: (annotation, target) ->
+          new TestAnchor am, annotation, target,
+            target.startPage, target.endPage
+
+      am.chooseAccessPolicy()
+
+      am
+
+    # Helper function to create an annotation with several
+    # targets, search of them potentially targeting a given
+    # range of pages.
+    createTestAnnotationForPages = (id, pageRanges) ->
+      result =
+        id: "annotation " + id
+        target: []
+        anchors: []
+
+      index = 0
+      for targetRange in pageRanges
+        [start, end] = if Array.isArray targetRange
+          targetRange
+        else
+          [targetRange, targetRange]
+        result.target.push
+          id: "target " + id + "-" + index++
+          startPage: start
+          endPage: end
+
+      result
+
+
+
+    describe "createAnchor", ->
+
+      it 'adds the created anchors to the correct per-page array', ->
+        am = createAnchoringManagerAndLazyDocument()
+        ann = createTestAnnotationForPages "a1", [1,[2,3]]
+        anchor1 = am.createAnchor(ann, ann.target[0]).result
+        anchor2 = am.createAnchor(ann, ann.target[1]).result
+        assert.include am.anchors[1], anchor1
+        assert.notInclude am.anchors[2], anchor1
+        assert.notInclude am.anchors[3], anchor1
+
+        assert.notInclude am.anchors[1], anchor2
+        assert.include am.anchors[2], anchor2
+        assert.include am.anchors[3], anchor2
+
+    it 'creates real anchors (complete with highlights), if the wanted page is already rendered', ->
+      am = createAnchoringManagerAndLazyDocument()
+      assert am.annotator.publish.notCalled
+      am.document.renderPage 1
+      ann = createTestAnnotationForPages "a1", [1]
+      anchor = am.createAnchor(ann, ann.target[0]).result
+      assert anchor.fullyRealized
+      assert.isObject anchor.highlight
+      hl = anchor.highlight[1]
+      assert.ok hl
+      assert.equal hl.page, 1
+      assert am.annotator.publish.calledWith "highlightsCreated", [hl]
+
+    it 'creates virtual anchors (that are not realized, thus have no highlights), when the wanted page is not rendered', ->
+      am = createAnchoringManagerAndLazyDocument()
+      ann = createTestAnnotationForPages "a1", [1]
+      anchor = am.createAnchor(ann, ann.target[0]).result
+
+      # The anchor is aware that it's incomplete
+      assert !anchor.fullyRealized
+
+      # At this point, no pages are rendered
+      assert.isObject anchor.highlight
+      assert.notOk anchor.highlight[1], "Should not have a highlight on page 1"
+
+      assert am.annotator.publish.notCalled
```

Comment:
Repetition in tests is fine if it improves readability for others coming to the tests later on and clarity of errors when running the tests.

My general rule of thumb is to repeat setup in the method if its just a few lines, otherwise look at ways of moving it into a setup helper. The more contained the test the better, so it's a matter of balance.